src/main/java/org/springframework/data/hazelcast/repository/query/GeoPredicate.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.hazelcast.repository.query;
+
+import static com.hazelcast.query.impl.IndexUtils.canonicalizeAttribute;
+
+import java.util.Map;
+
+import org.springframework.data.geo.Distance;
+import org.springframework.data.geo.Metric;
+import org.springframework.data.geo.Metrics;
+import org.springframework.data.geo.Point;
+
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.impl.Extractable;
+/**
+ * Geo Predicate
+ * @param <K> key of map entry
+ * @param <V> value of map entry
+ * @author Ulhas R Manekar
+ */
+public class GeoPredicate<K, V>
+        implements Predicate<K, V> {
+
+    private static final long serialVersionUID = 2873594766714576764L;
+    private static final double MILES_TO_KM = 1.609344;
+    private static final double MILES_TO_NUTRAL = 0.8684;
+    private static final double MINUTES_IN_DEGREE = 60;
+    private static final double STATUTE_MILES_IN_NAUTICAL_MILE = 1.1515;
+
+    final String attributeName;
+    final Point queryPoint;
+    final Distance distance;
+
+    public GeoPredicate(String attribute, Point point, Distance distance) {
+        this.attributeName = canonicalizeAttribute(attribute);
```

Comment:
> readAttributeValue

Its not possible, there are multiple obstacles
1. readAttributeValue method is private, no visibility in sub class.
2. attributeName - this variable also not visible in sub class.
3. I can't reply on apply() method of AbstractPredicate, because it eventually calls applyForSingleAttributeValue(Comparable attributeValue), which I need to provide implementation in sub class,  but catch is **Point is not a Comparable**, it throws cast exception.

So with all these points just for sake of extending AbstractPredicate doesn't help in any way. Thoughts ?